### PR TITLE
Add spec tests to microstack refresher spec

### PR DESCRIPTION
**Pull request by agrare**
Move the keystone v3 and unsupported version specs to the microstack refresher_spec.rb

This is part of the effort to move all spec tests to the microstack based VCRs rake tests so that they can be reliably re-recorded by anyone started in https://github.com/ManageIQ/manageiq-providers-openstack/pull/814

### CODE REVIEW

File: spec/models/manageiq/providers/openstack/cloud_manager/refresher_spec.rb [hunk 1]
```diff
@@ -6,24 +6,82 @@
   end
 
   context "#refresh" do
-    before(:each) do
```
Comment (kbrock) [inline spec/models/manageiq/providers/openstack/cloud_manager/refresher_spec.rb:9]: yea, think you wanted to keep this line

**Comment by agrare** [comment]
> yea, think you wanted to keep this line

No I thought it was better to use `let()` instead of a before and using 
ivars so that we could control the parameters easier

**Comment by agrare** [comment]
@kbrock can you take a look?

**Comment by agrare** [comment]
@kbrock :bow: thanks!

**Pull request merged**